core/camel-support/src/main/java/org/apache/camel/support/resume/AdapterHelper.java

```diff
@@ -41,9 +42,9 @@ private AdapterHelper() {
     }
 
     public static ResumeAdapter eval(CamelContext context, ResumeAware resumeAware, ResumeStrategy resumeStrategy) {
-        assert context != null;
-        assert resumeAware != null;
-        assert resumeStrategy != null;
+        ObjectHelper.notNull(context, "context");
+        ObjectHelper.notNull(resumeAware, "resumeAware");
+        ObjectHelper.notNull(resumeStrategy, "resumeStrategy");
```

Comment:
Assert is the correct one here as this is an internal interface where the nullity condition should not happen in any circumstance.

Reply:
My same reasoning as very top 👆.

Other than that when we run Camel's own tests (where Java assertion through surefire/failsafe plugins are enabled) then it could be that all these assertions are properly fulfilled which is given through a **concrete test setup**, but what if the setup and data of a given Camel user's production environment is different than what we had by our tests which suddenly they would end up with null values here. So using `ObjectHelper.notNull` guarantees we always verify these, no matter if through Camel test **setup and configuration** or the ones from the users.

Reply:
The reasoning is the other way around: is to ensure that we catch misbehaved tests, tests that may affect this piece of code with these or any thing non-production that may be doing something nasty. 

The nullity check for most of these is already performed by the calling code. This one is a good example of that:
- [MasterConsumer.java](https://github.com/apache/camel/blob/main/components/camel-master/src/main/java/org/apache/camel/component/master/MasterConsumer.java#L143-L152)
- [DefaultRoute.java](https://github.com/apache/camel/blob/main/core/camel-base-engine/src/main/java/org/apache/camel/impl/engine/DefaultRoute.java#L646-L649)

So, the native assertion allow us to add a small layer of "protection" without adding too many runtime checks.